account_bank_receipt/models/account_bank_receipt.py

```diff
@@ -0,0 +1,378 @@
+    )
+    currency_id = fields.Many2one(
+        comodel_name='res.currency',
+        required=True,
+        states={'done': [('readonly', '=', True)]},
+    )
+    currency_none_same_company_id = fields.Many2one(
+        comodel_name='res.currency',
+        string='Currency (False if same as company)',
+        compute='_compute_bank_receipt',
+        store=True,
+    )
+    state = fields.Selection([
+        ('draft', 'Draft'),
+        ('done', 'Done')],
+        string='Status',
+        default='draft',
+        readonly=True,
+    )
+    move_id = fields.Many2one(
+        comodel_name='account.move',
+        string='Journal Entry',
+        readonly=True,
+    )
+    bank_journal_id = fields.Many2one(
+        comodel_name='account.journal',
+        string='Bank Account',
+        required=True,
+        domain="[('company_id', '=', company_id), ('type', '=', 'bank'), "
+               "('bank_account_id', '!=', False)]",
+        states={'done': [('readonly', '=', True)]}
+    )
+    line_ids = fields.One2many(
+        comodel_name='account.move.line',
+        related='move_id.line_ids',
+        string='Lines',
+        readonly=True,
+    )
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        required=True,
+        states={'done': [('readonly', '=', True)]},
+        default=lambda self: self.env['res.company']._company_default_get(),
+    )
+    total_amount = fields.Float(
+        compute='_compute_bank_receipt',
+        readonly=True,
+        store=True,
+        digits=dp.get_precision('Account'),
+    )
+    check_count = fields.Integer(
+        string='Number of Checks',
+        compute='_compute_bank_receipt',
+        readonly=True,
+        store=True,
+    )
+    is_reconcile = fields.Boolean(
+        string='Reconcile',
+        compute='_compute_bank_receipt',
+        readonly=True,
+        store=True,
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        domain="[('customer', '=', True)]",
+        required=True,
+        states={'done': [('readonly', True)]},
+    )
+    receipt_line = fields.One2many(
+        comodel_name='account.bank.receipt.line',
+        inverse_name='order_id',
+        states={'done': [('readonly', True)]},
+    )
+
+    @api.constrains('receipt_line')
+    def _check_allocation(self):
+        for rec in self:
+            payments = rec.receipt_line.mapped('move_line_ids')
+            for payment in payments:
+                allocation = sum(rec.receipt_line.filtered(
+                    lambda l: l.move_line_ids.id == payment.id
+                    ).mapped('allocation'))
+                residual = payment.amount_residual
+                if allocation > residual:
+                    raise ValidationError(_(
+                        '%s has sum Allocation > Residual'
+                        ) % (payment.move_id.name))
+
+    def unlink(self):
+        for rec in self:
```

Comment:
Shorten code by
`if 'done' in self.mapped('state'):`

Reply:
it's shorten but sentence can't show name when selected more than 1